capstone/src/main/java/com/google/step/servlets/RestaurantServlet.java

```diff
@@ -0,0 +1,137 @@
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for sending/getting a restaurant to/from Datastore. */
+@WebServlet("/restaurant")
+public class RestaurantServlet extends HttpServlet {
+  private RestaurantClient restaurantClient = new RestaurantClient();
+
+  /**
+   * Will put the information gathered from the signup page and add the
+   * information into their respective properties for .
+   * @param request sent by the frontend. Requires the user to be logged in
+   *                and have the following parameters in the request body:
+   *                "name": name of the restaurant,
+   *                "cuisine": list of cuisines of the restaurant as CSV,
+   *                "story": story of the restaurant,
+   *                "phone": cellphone of the restaurant,
+   *                "website": link to their website,
+   * @param response will redirect to the home page of the website. If the
+   *                 user is not logged in when requesting the POST, the
+   *                 response will have a SC_FORBIDDEN status.
+   * @throws IOException
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Only available if the user is logged in.
+    if (!userService.isUserLoggedIn()) {
+      response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    String email = userService.getCurrentUser().getEmail();
+
+    String name = request.getParameter("name");
+
+    // Split the cuisine into a list
+    String cuisineString = request.getParameter("cuisine");
+    List<String> cuisineList = Arrays.asList(cuisineString.split(","));
+
+    String story = request.getParameter("story");
+
+    // Make sure its only numbers
+    String phone = request.getParameter("phone");
+    phone = changeNonNumericCharacters(phone);
+
+    // TODO: Make sure its a link.
+    String website = request.getParameter("website");
+
+    // Hard coded while we don't have Maps API functionality
+    GeoPt geoPoint = new GeoPt((float) 42.23422, (float) -87.234987);
+
+    // The following value is hardcoded while we implement the properties.
+    String status = "OKAY";
+
+    Restaurant restaurant =
+        new Restaurant(null, name, geoPoint, story, cuisineList, phone, website, status);
+
+    restaurantClient.putRestaurant(restaurant, email);
+  }
+
+  private String changeNonNumericCharacters(String num) {
+    return num.replaceAll("[^0-9]", "");
+  }
+
+  /**
+   * Returns a single Restaurant based on restaurantKey.
+   * @param response All details for the requested restaurant, in the following json format:
+  {
+  "restaurant" : {
+  "restaurantKey": long,
+  "name": String,
+  "location": GeoPt,
+  "story": String,
+  "cuisine": List<String>,
+  "phone": String,
+  "website": String,
+  "status": String
+  }
+  }
```

Comment:
Nit: fix spacing

Reply:
I fixed this change before creating PR #24 .